packages/fxa-settings/src/pages/Signin/SigninUnblock/container.tsx

```diff
@@ -44,22 +49,23 @@ const SigninUnblockContainer = ({
   const wantsTwoStepAuthentication =
     isOAuthIntegration(integration) && integration.wantsTwoStepAuthentication();
 
+  const { finishOAuthFlowHandler, oAuthDataError } = useFinishOAuthFlowHandler(
+    authClient,
+    integration
+  );
+
   const [beginSignin] = useMutation<BeginSigninResponse>(BEGIN_SIGNIN_MUTATION);
 
   const signinWithUnblockCode: BeginSigninWithUnblockCodeHandler = async (
     unblockCode: string
   ) => {
-    // TODO in FXA-6518 oauth ticket
-    // const service = integration.getService();
+    const service = integration.getService();
     const options = {
       verificationMethod: VerificationMethods.EMAIL_OTP,
+      keys: integration.wantsKeys(),
+      ...(service !== MozServices.Default && { service }),
```

Comment:
huh what happens when `service` is `MozServices.Default`?

Reply:
Good question - I'd added this to SigninUnblock to match updates by Lauren to the Signin page. Looks like it might be to match parity with content server - service is only included for signin/signup when it is a relying party and is used to notify users that a new service has been attached to their account.

Reply:
Oh sorry, I meant syntactically you cannot spread a false, so wouldn't that result in a syntax error?

Reply:
Ah, we use this syntax in multiple places in the codebase, and it seems to be supported according to the "Conditionally adding properties to an object" section of [MDN's docs on spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax)

>You can make an element present or absent in an object literal, depending on a condition, using a conditional operator.
(...)
> Because primitives can be spread into objects as well, and from the observation that all falsy values do not have enumerable properties, you can simply use a logical AND operator:

```
const isSummer = false;
const fruits = {
  apple: 10,
  banana: 5,
  ...(isSummer && { watermelon: 30 }),
};
```